archinstall/lib/bootloader/bootloader_menu.py

```diff
@@ -0,0 +1,236 @@
+import textwrap
+from typing import override
+
+from archinstall.lib.translationhandler import tr
+from archinstall.tui.curses_menu import SelectMenu
+from archinstall.tui.menu_item import MenuItem, MenuItemGroup
+from archinstall.tui.result import ResultType
+from archinstall.tui.types import Alignment, FrameProperties, Orientation
+
+from ..args import arch_config_handler
+from ..hardware import SysInfo
+from ..menu.abstract_menu import AbstractSubMenu
+from ..models.bootloader import Bootloader, BootloaderConfiguration
+
+
+class BootloaderMenu(AbstractSubMenu[BootloaderConfiguration]):
+	def __init__(
+		self,
+		bootloader_conf: BootloaderConfiguration,
+	):
+		self._bootloader_conf = bootloader_conf
+		menu_options = self._define_menu_options()
+
+		self._item_group = MenuItemGroup(menu_options, sort_items=False, checkmarks=True)
+		super().__init__(
+			self._item_group,
+			config=self._bootloader_conf,
+			allow_reset=False,
+		)
+
+	def _define_menu_options(self) -> list[MenuItem]:
+		bootloader = self._bootloader_conf.bootloader
+		has_uefi = SysInfo.has_uefi()
+
+		# UKI availability
+		uki_enabled = has_uefi and bootloader.has_uki_support()
+		if not uki_enabled:
+			self._bootloader_conf.uki = False
+
+		# Removable availability
+		removable_enabled = has_uefi and bootloader.has_removable_support()
+		if not removable_enabled:
+			self._bootloader_conf.removable = False
+
+		return [
+			MenuItem(
+				text=tr('Bootloader'),
+				action=self._select_bootloader,
+				value=self._bootloader_conf.bootloader,
+				preview_action=self._prev_bootloader,
+				mandatory=True,
+				key='bootloader',
+			),
+			MenuItem(
+				text=tr('Unified kernel images'),
+				action=self._select_uki,
+				value=self._bootloader_conf.uki,
+				preview_action=self._prev_uki,
+				key='uki',
+				enabled=uki_enabled,
+			),
+			MenuItem(
+				text=tr('Install to removable location'),
+				action=self._select_removable,
+				value=self._bootloader_conf.removable,
+				preview_action=self._prev_removable,
+				key='removable',
+				enabled=removable_enabled,
+			),
+		]
+
+	def _prev_bootloader(self, item: MenuItem) -> str | None:
+		if item.value:
+			return f'{tr("Bootloader")}: {item.value.value}'
+		return None
+
+	def _prev_uki(self, item: MenuItem) -> str | None:
+		if item.value:
+			return tr('Enabled')
+		return tr('Disabled')
+
+	def _prev_removable(self, item: MenuItem) -> str | None:
+		if item.value:
+			return tr('Will install to /EFI/BOOT/ (removable location)')
+		return tr('Will install to standard location with NVRAM entry')
+
+	@override
+	def run(
+		self,
+		additional_title: str | None = None,
+	) -> BootloaderConfiguration:
+		super().run(additional_title=additional_title)
+		return self._bootloader_conf
+
+	def _select_bootloader(self, preset: Bootloader | None) -> Bootloader | None:
```

Comment:
can we move this outside the class as a standalone function similar to the ask* functions. Reason being they can be reused if someone wants to build their own menu and just use these selections

Reply:
but that function depends on `self` because its purpose is to ensure the submenu's UKI/removable options are dynamically updated depending on the chosen bootloader.